### PR TITLE
Add junit Jest reporter

### PR DESCRIPTION
**Pull request by pdehaan**
# References:

Jira: MNTOR-
Figma:

# Description
Early work on adding JUnit XML reports for the Jest unit tests. Eventually we'll want to ingest these into a pipeline to track tests and test performance over time (which means working on getting these stored as artifacts in the GitHub Actions pipeline too).

# Screenshot (if applicable)

Not applicable.

# How to test

Run tests locally, look for a root /jest-junit.xml file.

# Checklist (Definition of Done)

- [ ] Localization strings (if needed) have been added.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] If this PR implements a feature flag or experimentation, the Ship Behind Feature Flag status in Jira has been set
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met.
- [ ] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: jest.config.cjs [hunk 1]
```diff
@@ -51,12 +51,12 @@ const customJestConfig = {
...
-  // coverageReporters: [
-  //   "json",
-  //   "text",
-  //   "lcov",
-  //   "clover"
-  // ],
+  coverageReporters: [
+    // "json",
+    "text",
+    // "lcov",
```
Comment (Vinnl) [inline jest.config.cjs:57]: I think this is the one that generates the HTML report that I use to check if there's any code I forgot to write tests for:

```suggestion
     "lcov",
```

**Review by Vinnl — APPROVED** [review]
I believe on Relay we use coveralls.io, maybe we want to use that here too, to save the effort of setting up a full pipeline?

**Comment by rhelmer** [comment]
@pdehaan are you still working on this one?

**Comment by Vinnl** [comment]
I'm assuming no, so closing for now :)

**Pull request closed without merging**